node/src/types/deploy.rs

```diff
@@ -827,27 +828,380 @@ impl Deploy {
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            payment,
+            self.session().clone(),
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_missing_entry_point_in_payment_contract(
+        &mut self,
+        rng: &mut TestRng,
+    ) -> Self {
+        let payment = ExecutableDeployItem::StoredContractByName {
+            name: "Test".to_string(),
+            entry_point: "non-existent-entry-point".to_string(),
+            args: Default::default(),
+        };
+        let secret_key = SecretKey::random(rng);
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            payment,
+            self.session().clone(),
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_valid_custom_payment_package_by_name(&mut self, rng: &mut TestRng) -> Self {
+        let payment = ExecutableDeployItem::StoredVersionedContractByName {
+            name: "Test".to_string(),
+            version: None,
+            entry_point: "call".to_string(),
+            args: Default::default(),
+        };
+
+        let secret_key = SecretKey::random(rng);
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            payment,
+            self.session().clone(),
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_missing_payment_package_by_hash(&mut self, rng: &mut TestRng) -> Self {
+        let payment = ExecutableDeployItem::StoredVersionedContractByHash {
+            hash: Default::default(),
+            version: None,
+            entry_point: "call".to_string(),
+            args: Default::default(),
+        };
+        let secret_key = SecretKey::random(rng);
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            payment,
+            self.session().clone(),
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_missing_contract_version_in_payment_package(
+        &mut self,
+        rng: &mut TestRng,
+    ) -> Self {
+        let payment = ExecutableDeployItem::StoredVersionedContractByName {
+            name: "Test".to_string(),
+            version: Some(6u32),
```

Comment:
`version` in `payment` is `Some`, yet the function name says `with_missing_contract_version_in_payment...`, maybe it's just a naming issue.

Reply:
Renamed the function to say `non-existent` instead of missing